libs/platform/src/lib/components/form/form-group/form-group.component.ts

```diff
@@ -348,126 +418,43 @@ export class FormGroupComponent implements FormGroupContainer, OnInit, AfterCont
-                        right[indexR].columns
-                    );
-
-                    f.styleClass = `col-sm-${f.columns} col-md-${f.columns} col-lg-${f.columns}`;
-                    merged[current++] = f;
-                    indexR++;
-                }
-            }
-            return merged;
-        } else if (left.length > 0) {
-            // when only one column dont use 6 columsn and 6 colums
-            return left.map((item) => {
-                item.isFluid = true;
-                return item;
-            });
-        }
-        return [];
-    }
-
-    /**
-     * This is just the temp solution until I can figure out better way of doing this. If I want
-     * to have e.g. 3 fields on the left and 1 field on the right. It will not work with current
-     * 6-column width layout as left and right side are not even.
-     *
+     * @hidden
+     * if `columnLayoutType` is given, set those column layouts appropriately
      */
-    private evenFields(zLeft: GroupField[], zRight: GroupField[]): void {
-        zLeft.sort((a, b) => a.rank - b.rank);
-
-        if (zRight.length === 0) {
-            return;
-        }
-
-        zRight.sort((a, b) => a.rank - b.rank);
-        if (zLeft.length !== zRight.length) {
-            // retrieve the smallest from both
-            const toEven = zLeft.length > zRight.length ? zRight : zLeft;
-            for (let i = 0; i <= Math.abs(zLeft.length - zRight.length); i++) {
-                const zone = toEven[0].zone;
-                toEven.push(new GroupField(zone, `${zone}-${i}`, toEven.length + 1, null, 6));
+    private _setUserSpecifiedLayout(groupField?: GroupField): void {
+        const [xl, lg, md] = this.columnLayoutType.split('-');
+        const xlColumnsNumber = parseInt(xl.slice(2, xl.length), 10);
+        const lgColumnsNumber = parseInt(lg.slice(1, lg.length), 10);
+        const mdColumnsNumber = parseInt(md.slice(1, md.length), 10);
+
+        if (isNaN(xlColumnsNumber) || isNaN(lgColumnsNumber) || isNaN(mdColumnsNumber)) {
+            throw new Error('Input a valid number for columns');
+        } else if (xlColumnsNumber > 12 || lgColumnsNumber > 12 || mdColumnsNumber > 12) {
+            throw new Error('Columns cannot be more than 12');
+        } else {
+            const lgColumns = 12 / lgColumnsNumber;
+            const mdColumns = 12 / mdColumnsNumber;
+            const xlColumns = 12 / xlColumnsNumber;
+
+            // for `lg` single-column layout, Styles does not use any class, and providing `fd-col-lg--12` has unintended side-effects
+            // therefore, we remove the lg class for single-column layout
+            if (lgColumns === 12) {
+                this.xlCol = `fd-col-xl--${xlColumns} fd-col-md--${mdColumns}`;
+            } else {
+                this.xlCol = `fd-col-xl--${xlColumns} fd-col-md--${mdColumns} fd-col-lg--${lgColumns}`;
             }
         }
-        return;
     }
 }
 
 export class GroupField {
```

Comment:
GroupField was that as internal structure to keep fields with some additional metadata used for rendering. How our FiledGroup fits (the heading we have)

Reply:
Children of `FormFieldGroup` wrapped `GroupField` as well. The heading stored separately